src/Microsoft.DotNet.Build.Tasks/ComputeDestinationsForDependencies.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class ComputeDestinationsForDependencies : Task
+    {
+        [Required]
+        public ITaskItem[] TestDependencies
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] TestDependenciesWithDestinations
+        {
+            get { return TestDependencies; }
+        }
+
+        public override bool Execute()
+        {
+            foreach (ITaskItem dependency in TestDependencies)
+            {
+                string relativeDestinationPath;
+
+                bool? preserveSubDirectories = dependency.GetMetadata("PreserveSubDirectories")?.Equals("true", StringComparison.OrdinalIgnoreCase);
+                if (preserveSubDirectories == true)
+                {
+                    string packageRelativePath = dependency.GetMetadata("PackageRelativePath");
+
+                    // PackageRelativePath contains (PackageName\VersionNumber\[Directories\]FileName). This is to remove the first two directories on 
+                    // the path to preserve just the directory structure.
+                    string[] segments = packageRelativePath.Split(new Char[] { '/', '\\'});
```

Comment:
Is it possible that packageRelativePath is null, which would cause a null ref exception here?

` <PackageRelativePath Condition="'%(TestCopyLocal.NugetPackageId)' != ''">`